extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -360,18 +347,19 @@ export class PublishDatabaseDialog {
 		}).component();
 
 		editConnectionButton.onDidClick(async () => {
-			this.connection = await this.apiWrapper.openConnectionDialog();
+			let connection = await this.apiWrapper.openConnectionDialog();
```

Comment:
Since the connection string can be read from the profile, depending on the authentication, connectionConnect() or 				openConnectionDialog() will get called. openConnectionDialog() returns a connection, but connectionConnect() returns ConnectionResult which only has connectionId. The getUriForConnection() used to get the connection uri for publishing also only needed the connectionId anyway.